pkg/restore/restore.go

```diff
@@ -979,16 +983,29 @@ func (ctx *restoreContext) restoreItem(obj *unstructured.Unstructured, groupReso
 	name := obj.GetName()
 
 	// Check if we've already restored this itemKey.
-	itemKey := velero.ResourceIdentifier{
-		GroupResource: groupResource,
-		Namespace:     namespace,
-		Name:          name,
+	itemKey := itemKey{
+		resource:  resourceKey(obj),
+		namespace: namespace,
+		name:      name,
 	}
 	if _, exists := ctx.restoredItems[itemKey]; exists {
 		ctx.log.Infof("Skipping %s because it's already been restored.", resourceID)
 		return warnings, errs
 	}
-	ctx.restoredItems[itemKey] = struct{}{}
+	ctx.restoredItems[itemKey] = ""
+	defer func() {
+		// the action field is set explicitly
+		if action := ctx.restoredItems[itemKey]; len(action) > 0 {
```

Comment:
Looks like this is the only place to check the value of `ctx.restoredItems`, and it checks if the value is empty or not. Therefore, do we need to define these detailed strings as below?
```
	itemRestoreResultCreated = "created"
	itemRestoreResultUpdated = "updated"
	itemRestoreResultFailed  = "failed"
	itemRestoreResultSkipped = "skipped"
```